src/sentry/static/sentry/app/views/organizationIntegrations/integrationRepos.tsx

```diff
@@ -17,27 +17,45 @@ import RepositoryRow from 'app/components/repositoryRow';
 import {Panel, PanelBody, PanelHeader} from 'app/components/panels';
 import space from 'app/styles/space';
 import {t} from 'app/locale';
+import {Integration, Repository} from 'app/types';
 
-export default class IntegrationRepos extends AsyncComponent {
+type Props = AsyncComponent['props'] & {
+  integration: Integration;
+};
+
+type State = AsyncComponent['state'] & {
+  adding: boolean;
+  dropdownBusy: boolean;
+  itemList: Repository[];
+  integrationRepos: {
+    repos: {identifier: string; name: string}[];
+    searchable: boolean;
+  };
+};
+
+export default class IntegrationRepos extends AsyncComponent<Props, State> {
   static propTypes = {
     integration: PropTypes.object.isRequired,
   };
   static contextTypes = {
     organization: PropTypes.object.isRequired,
+    router: PropTypes.object,
   };
 
   constructor(props, context) {
```

Comment:
@nolachen Do we actually need the constructor here? I think it would make more sense to just override the `getDefaultState` with the fields that are needed.